lib/linebuffer.js

```diff
@@ -21,18 +20,43 @@ var bufferStats = {
 };
 var config = require('./config');
 var dccount = 0;
-var httperr;
+var httpErrorTime;
 var queue;
 
+// check response to see need for buffering into queue
+const needsRetry = (error, response, body) => {
+    if (error) {
+        debug(`req error: ${error}`);
+        if (!httpErrorTime) { httpErrorTime = Date.now(); }
+        if (bufferStats.flushcount % 10 === 0) { utils.log(`request error: ${error}`); }
+        return config.HTTP_RETRY && config.DEFAULT_HTTP_ERRORS && config.DEFAULT_HTTP_ERRORS.includes(error.code);
+    }
+
+    if (response) {
+        if (response.statusCode !== 200) {
+            debug(`req failed: ${body}`);
+            if (bufferStats.flushcount % 10 === 0) { utils.log(`request failed: ${body}`); }
+            return response.statusCode >= 500;
+        } else {
+            if (httpErrorTime) {
+                debug('req succeeded, re-enabling');
+                httpErrorTime = null;
+            }
+        }
+    }
+
+    return false;
+};
+
 const flush = () => {
     config.flushtimeout = setTimeout(flush, config.FLUSH_INTERVAL);
     if (buf.length === 0) return;
-    if (httperr && config.TRANSPORT === 'http') {
-        if (Date.now() - httperr < config.HTTP_TIMEOUT / 2) {
+    if (httpErrorTime && config.TRANSPORT === 'http') {
+        if (Date.now() - httpErrorTime < config.HTTP_TIMEOUT / 2) {
             dccount = buf.length;
             return;
         } else {
-            httperr = Date.now(); // reset
+            httpErrorTime = Date.now(); // reset
             debug('req re-attempting');
         }
```

Comment:
No need for the above `else` block, as the `if` block already `return`

Reply:
Done!